includes/api/save-abandonment-data.php

```diff
@@ -291,6 +298,433 @@ function saveWooCartAbandonmentRecoveryData($razorpayData)
     return $logObj;
 }
 
+//Save cart abandonment data for CartBounty plugin
+function saveCartBountyData($razorpayData){
+    global $wpdb;
+    $products        = WC()->cart->get_cart();
+    $ghost           = false;
+    $name            = $razorpayData['customer_details']['billing_address']['name'] ?? '';
+    $surname         = " ";
+    $email           = $razorpayData['customer_details']['email'];
+    $phone           = $razorpayData['customer_details']['contact'];
+    $cart_table      = $wpdb->prefix ."cartbounty";
+    $cart            = readCartCB($razorpayData['receipt']);
+   
+    $location = array(
+      'country' 	=> $razorpayData['customer_details']['shipping_address']['country'] ?? '',
+      'city' 		=> $razorpayData['customer_details']['shipping_address']['city'] ?? '',
+      'postcode' 	=> $razorpayData['customer_details']['shipping_address']['zipcode'] ?? ''
+  );
+
+  $other_fields = array(
+      'cartbounty_billing_company' 		=> '',
+      'cartbounty_billing_address_1'    => $razorpayData['customer_details']['billing_address']['line1'] ?? '',
+      'cartbounty_billing_address_2' 	=> $razorpayData['customer_details']['billing_address']['line2'] ?? '',
+      'cartbounty_billing_state' 		=> $razorpayData['customer_details']['billing_address']['state'] ?? '',
+      'cartbounty_shipping_first_name' 	=> $razorpayData['customer_details']['billing_address']['name'] ?? '',
+      'cartbounty_shipping_last_name'   => '',
+      'cartbounty_shipping_company' 	=> '',
+      'cartbounty_shipping_country' 	=> $razorpayData['customer_details']['shipping_address']['country'] ?? '',
+      'cartbounty_shipping_address_1'   => $razorpayData['customer_details']['shipping_address']['line1'] ?? '',
+      'cartbounty_shipping_address_2'   => $razorpayData['customer_details']['shipping_address']['line2'] ?? '',
+      'cartbounty_shipping_city' 		=> $razorpayData['customer_details']['shipping_address']['city'] ?? '',
+      'cartbounty_shipping_state' 		=> $razorpayData['customer_details']['shipping_address']['state'] ?? '',
+      'cartbounty_shipping_postcode' 	=> $razorpayData['customer_details']['shipping_address']['zipcode'] ?? '',
+      'cartbounty_order_comments' 	    => '',
+      'cartbounty_create_account' 	    => '',
+      'cartbounty_ship_elsewhere' 		=> ''
+  );
+
+  $user_data = array(
+      'name'			=> $name,
+      'surname'		    => $surname,
```

Comment:
@angbpy formatting the code.

Reply:
@ChetanGN This is actually correctly formatted , it looks this way on github